core/trino-main/src/test/java/io/trino/sql/planner/optimizations/TestPartialTopNWithPresortedInput.java

```diff
@@ -0,0 +1,188 @@
+import static io.trino.sql.planner.plan.ExchangeNode.Scope.REMOTE;
+import static io.trino.sql.planner.plan.ExchangeNode.Type.GATHER;
+import static io.trino.sql.planner.plan.ExchangeNode.Type.REPARTITION;
+import static io.trino.sql.planner.plan.TopNNode.Step.FINAL;
+import static io.trino.sql.planner.plan.TopNNode.Step.PARTIAL;
+import static io.trino.sql.tree.SortItem.NullOrdering.FIRST;
+import static io.trino.sql.tree.SortItem.NullOrdering.LAST;
+import static io.trino.sql.tree.SortItem.Ordering.ASCENDING;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+
+public class TestPartialTopNWithPresortedInput
+        extends BasePlanTest
+{
+    private static final String MOCK_CATALOG = "mock_catalog";
+    private static final String TEST_SCHEMA = "test_schema";
+
+    private static final SchemaTableName tableA = new SchemaTableName(TEST_SCHEMA, "table_a");
+    private static final String columnNameA = "col_a";
+    private static final ColumnHandle columnHandleA = new MockConnectorColumnHandle(columnNameA, VARCHAR);
+    private static final String columnNameB = "col_b";
+
+    @Override
+    protected LocalQueryRunner createLocalQueryRunner()
+    {
+        Session session = testSessionBuilder()
+                .setCatalog(MOCK_CATALOG)
+                .setSchema(TEST_SCHEMA)
+                .build();
+        LocalQueryRunner queryRunner = LocalQueryRunner.builder(session).build();
+        MockConnectorFactory mockFactory = MockConnectorFactory.builder()
+                .withGetTableProperties((connectorSession, handle) -> {
+                    MockConnectorTableHandle tableHandle = (MockConnectorTableHandle) handle;
+                    if (tableHandle.getTableName().equals(tableA)) {
+                        return new ConnectorTableProperties(
+                                TupleDomain.all(),
+                                Optional.empty(),
+                                Optional.empty(),
+                                Optional.empty(),
+                                ImmutableList.of(new SortingProperty<>(columnHandleA, ASC_NULLS_FIRST)));
+                    }
+                    throw new IllegalArgumentException();
+                })
+                .withGetColumns(schemaTableName -> {
+                    if (schemaTableName.equals(tableA)) {
+                        return ImmutableList.of(
+                                new ColumnMetadata(columnNameA, VARCHAR),
+                                new ColumnMetadata(columnNameB, VARCHAR));
+                    }
+                    throw new IllegalArgumentException();
+                })
+                .build();
+        queryRunner.createCatalog(MOCK_CATALOG, mockFactory, ImmutableMap.of());
+        return queryRunner;
+    }
+
+    @Test
+    public void testWithSortedTable()
+    {
+        ImmutableList<PlanMatchPattern.Ordering> orderBy = ImmutableList.of(sort("t_col_a", ASCENDING, FIRST));
```

Comment:
Use `List<...>` for the declaration here and in other tests.

Reply:
done